Add PDS Banner App Bar

eddiesarevalo opened the pull request:
The PDS banner app bar code has been added with keyboard accessibility support.

This banner app bar is meant to run independently of the WDS but will be included for use in React + Material UI applications.

Instructions on how to use it are in the readme.td file in the components file directory pds-wds-react/components/
https://github.com/NASA-PDS/pds-wds-react/tree/pds-banner-app-bar/components

Resolves #1 
Resolves NASA-PDS/PDS.nasa.gov-Search#15
Resolves NASA-PDS/PDS.nasa.gov-Search#16

jordanpadams commented:
@eddiesarevalo can we see this deployed somewhere?

eddiesarevalo commented:
> @eddiesarevalo can we see this deployed somewhere?

@jordanpadams You can see it at: https://pds-dev2.jpl.nasa.gov/searchui/demo/

Try using the keyboard as well.

jordanpadams commented:
@eddiesarevalo looks great. super nitpick, but hte Info Icon looks a little pixelated on my screen compared to what i see on pds.nasa.gov. Is that expected from the icon set we are using or is there something else going on?

mdrum commented:
I was able to copy it into the Context Browser with no issue. Looks good, apart from the info icon issue Jordan noted. The source image appears to just be 14x14px.

Could this component get packaged and registered to npm, so that it could remain an external dependency that can be independently updated?

eddiesarevalo commented:
> @eddiesarevalo looks great. super nitpick, but hte Info Icon looks a little pixelated on my screen compared to what i see on pds.nasa.gov. Is that expected from the icon set we are using or is there something else going on?

@jordanpadams It might be because I used an image for the icon. I think in the pds.nasa.gov site it is an SVG. I can change it to use the SVG as well to make it look better.

jordanpadams commented:
+1 to @mdrum's recommendation.

@tloubrieu-jpl can we create another ticket to have the app banner packaged and registered to NPM as a follow-on task to this one?

eddiesarevalo commented:
@mdrum @jordanpadams The plan is definitely to change it to an npm package but I was hoping to have at least a couple of components that use the new theme. This version banner has its styles in the component code because we don't have the theme yet. So when we add the new components they will mess with the directory order. I was hoping to sort that out before the npm package is created and published.

jordanpadams commented:
@eddiesarevalo copy. thanks.

jordanpadams commented:
@tariqksoliman @riverma @anilnatha if you could poke at this sometime today or over the weekend that would be great so we can merge first thing Monday

eddiesarevalo commented on components/Banner.js:
```diff
@@ -0,0 +1,389 @@
+        backgroundColor: 'black',
+        boxShadow: 'none',
+        borderRadius: '0',
+        padding: '5px 0 6px 5px',
```
What number do you think @edwardbarraza @tariqksoliman ? I can change it to whatever you think is good or I can just move up the svg icon by a couple pixels to make it center.

tariqksoliman replied:
If you are changing the main height, I'd go with 32px since that's what the Earth daacs are using and being a power of two, I have to like it.

eddiesarevalo commented on components/Banner.js:
```diff
@@ -0,0 +1,389 @@
+                        {anchorEl ? <ArrowDropUpIcon className={classes.arrowIcon}/> : <ArrowDropDownIcon className={classes.arrowIcon}/>}
+                    </Button>
+
+                    <StyledMenu
```
Looks like this is the default behavior for the down/up arrow key accessibility. When the material UI menu is opened it automatically selects the first entry.  I'll look deeper into this.

tariqksoliman commented on components/Banner.js:
```diff
@@ -0,0 +1,389 @@
+                                className={classes.menuLink}
+                                href='https://naif.jpl.nasa.gov/naif'
+                                rel='noopener'
+                                inputRef={this.textFieldRef}
```
This seems like an unused boilerplate property and it's throwing this warning for me:
`
Warning: React does not recognize the `inputRef` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `inputref` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
`

eddiesarevalo commented:
Made the following changes:
-Changed info icon image to an svg icon.
-Added font weight 400 to node label.
-Added vertical to transform origin for info menu.
-Changed app bar height to 32px.
-Fixed issue with first item in node menu list being focused on click.
-Removed inputRef from item link that was causing warning message to appear.

tariqksoliman reviewed (approved):
Looks good!

tariqksoliman commented on components/Banner.js:
```diff
@@ -0,0 +1,389 @@
+                            vertical: 'bottom',
+                            horizontal: 'left'
+                        }}
+                        transformOrigin={{
```
There's another transformOrigin.horizontal set to an erroneous pixel value at L316 too.

tariqksoliman commented:
Could the last `horizontal` at Banner.js#L319 be changed too like the other ones. It's not breaking anything but it does throw a big error in my setup.

jordanpadams commented:
@eddiesarevalo ⬆️

eddiesarevalo commented:
> Could the last `horizontal` at Banner.js#L319 be changed too like the other ones (I missed this one earlier in my review). It's not breaking anything functionality-wise but it does throw a big error message in my setup.

Sure, I'll remove it.
